### ISSUE TITLE
2 Issues

### ISSUE THREAD

**Issue by windowshopr** [agassi4013/Deep-Q-Learning-Roulette#1]
Running this script on Windows 10, Python 3.6. 

1. You have to rename the folder "unittest" to something else, because depending on what version of Numpy you have installed, numpy tries to import its own "unittest" folder to look for a "TestCase" dependency, so if this folder name is already defined, the script will look in there first, and then error out saying it can't find "TestCase". So I just changed the folder name to "unittest2" and it started training. But not before...

2. Comment out lines 334-339 of "r3.py". Can't load model files until you've created them locally. 

I imagine I'll run into an error somewhere during training from renaming that "unittest2" folder, but will address it when it happens.

Thanks!

**Comment by windowshopr** [comment]
Also, put this code under line 343 of "r3.py" (timesteps_per_episode = 80)

```
# Make "saved_models" directory if doesn't exist yet
import os
raw_path = os.path.join('save_models')
if not os.path.isdir(raw_path):
    print('\nMaking direcory: ' + raw_path)
    os.makedirs(raw_path)
```

This makes the "saved_models" folder for the user (if it doesn't already exist) before getting "directory doesn't exist" error.

**Comment by windowshopr** [comment]
Also, shouldn't `self._action_size = 38` be `self._action_size = 37`? You would have 37 possible actions on the single number model (place a bet on 0, or any other 1 of 36 numbers = 37)? or is it to allow the argmax to inclusively get action 37? Thanks!

**Comment by agassi4013** [comment]
> Also, put this code under line 343 of "r3.py" (timesteps_per_episode = 80)
> 
> ```
> # Make "saved_models" directory if doesn't exist yet
> import os
> raw_path = os.path.join('save_models')
> if not os.path.isdir(raw_path):
>     print('\nMaking direcory: ' + raw_path)
>     os.makedirs(raw_path)
> ```
> 
> This makes the "saved_models" folder for the user (if it doesn't already exist) before getting "directory doesn't exist" error.

Thanks for it, just skip that step cause it should be easy to debug.

**Comment by agassi4013** [comment]
> Also, shouldn't `self._action_size = 38` be `self._action_size = 37`? You would have 37 possible actions on the single number model (place a bet on 0, or any other 1 of 36 numbers = 37)? or is it to allow the argmax to inclusively get action 37? Thanks!

There is one more action to pass (no bets) given the observation state, it would make the training more difficult to converge but it is more reasonable to just observe and wait for more desirable patterns.
It is  mentioned on the medium article.

**Comment by windowshopr** [comment]
Right you are, I noticed that after I started playing with it a little bit. It's an interesting project, although what I found after training and testing the Single Number model, I printed out the action for each step in the test, and all it did was predict the number 35, regardless of the state, so it got stuck in just predicting a single value. But, I think there's lots of potential to work in betting progressions, predicting multiple numbers, etc. Good work!
